app/uk/gov/hmrc/bindingtariffclassification/repository/SearchMapper.scala

```diff
@@ -36,16 +36,20 @@ class SearchMapper {
         filter.traderName.map("application.holder.businessName" -> nullifyNoneValues(_)) ++
         filter.minDecisionEnd.map("decision.effectiveEndDate" -> greaterThan(_)(formatInstant)) ++
         filter.commodityCode.map("decision.bindingCommodityCode" -> numberStartingWith(_)) ++
-        filter.goodDescription.map("application.goodDescription" -> contains(_))
+        filter.goodDescription.map("application.goodDescription" -> contains(_)) ++
+        filter.keywords.map("keywords" -> containsAll(_))
     )
   }
 
-
   def sortBy(sort: Sort): JsObject = {
     Json.obj(toMongoField(sort.field) -> sort.direction.id)
   }
 
-  def greaterThan[T](value: T)(implicit writes: Writes[T]): JsObject = {
+  private def containsAll(strs: Set[String]): JsObject = {
+    Json.obj("$all" -> strs)
```

Comment:
https://docs.mongodb.com/manual/reference/operator/query/all/

Reply:
I think this needs to be an “Any” search rather than “All”

Reply:
No, the ticket description says:
```
Entering more keywords narrows the search (i.e. the search terms are joined by AND, not OR)
```